torch/nn/parallel/distributed.py

```diff
@@ -456,12 +456,15 @@ class DistributedDataParallel(Module, Joinable):
 
     .. note::
         DistributedDataParallel currently offers limited support for gradient
-        checkpointing with :meth:`torch.utils.checkpoint`. DDP will work as
-        expected when there are no unused parameters in the model and each layer
+        checkpointing with :meth:`torch.utils.checkpoint`.
+        If the checkpoint is done with use_reentrant=True (the default), DDP will work
+        as expected when there are no unused parameters in the model and each layer
         is checkpointed at most once (make sure you are not passing
         `find_unused_parameters=True` to DDP). We currently do not support the
         case where a layer is checkpointed multiple times, or when there unused
         parameters in the checkpointed model.
+        If, however, the checkpoint is done with use_reentrant=False, DDP will work
```

Comment:
I think we should change our documentation to officially recommend `use_reentrant=False` as that's the recommended way of `torch.utils.checkpoint`, and discourage `use_reentrant=True` usage. cc @rohan-varma

Reply:
I've updated the documentation